data_structures/src/Bag/Bag.cs

```diff
@@ -6,41 +6,80 @@ namespace ADT
 {
     public class Bag<T>: IBag<T>
     {
-        public Bag(int size = 10) { }
+        private T[] _bag { get; set; }
+        private int _fullSpots { get; set; }
+
+        public Bag(int size = 10) { 
+            _bag = new T[size];
```

Comment:
Since this is being passed in, you can use this to check if an add is legal. You will need to capture this in an instance variable, though. Maybe `_maxSize`.